reference_data/management/tests/update_dbnsfp_gene_2_3_tests.py

```diff
@@ -26,15 +28,15 @@ def setUp(self):
         # Create a temporary directory
         self.test_dir = tempfile.mkdtemp()
         self.temp_file_path = os.path.join(self.test_dir, 'dbNSFP_gene')
-        with open(self.temp_file_path, 'w') as f:
-            f.write(u''.join(dbNSFP_GENE_DATA))
+        with open(self.temp_file_path, 'wb') as f:
+            f.write(''.join(dbNSFP_GENE_DATA).encode('utf-8'))
 
     def tearDown(self):
         # Close the file, the directory will be removed after the test
         shutil.rmtree(self.test_dir)
 
     @mock.patch('reference_data.management.commands.utils.update_utils.logger')
-    @mock.patch('reference_data.management.commands.utils.download_utils.download_file')
+    @mock.patch('reference_data.management.commands.utils.update_utils.download_file')
```

Comment:
It is weird that when I mock `download_utils.download_file` the `download_file()` is not mocked and it downloads file from the network sometime, especially when I run all the tests for reference_data management commands.
I think it may be related to the order of test initialization and mock replacement. Sometime mocking fails because mocking happens after the object variables being initialized.
Mocking `update_utils.download_file` here is always correct because the `download_file()` is only called by `update_utils`.